samples/boot/saml2login/src/main/java/boot/saml2/config/Saml2SampleBootConfiguration.java

```diff
@@ -0,0 +1,188 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package boot.saml2.config;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.cert.X509Certificate;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.saml2.credentials.Saml2X509Credential;
+import org.springframework.security.saml2.serviceprovider.provider.InMemorySaml2RelyingPartyRepository;
+import org.springframework.security.saml2.serviceprovider.provider.Saml2RelyingPartyRegistration;
+import org.springframework.security.saml2.serviceprovider.provider.Saml2RelyingPartyRepository;
+import org.springframework.util.StringUtils;
+
+import boot.saml2.config.OpenSamlKeyConverters.Saml2X509CredentialConverter;
+
+import static java.util.Collections.emptyList;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialUsage.ENCRYPTION;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialUsage.VERIFICATION;
+
+@Configuration
+@ConfigurationProperties(prefix = "spring.security.saml2.login")
+@Import(OpenSamlKeyConverters.class)
+public class Saml2SampleBootConfiguration {
+
+	private List<IdentityProvider> providers;
+
+	@Bean
+	public Saml2RelyingPartyRepository saml2IdentityProviderDetailsRepository() {
+		return new InMemorySaml2RelyingPartyRepository(getIdentityProviders(providers));
+	}
+
+	public void setIdentityProviders(List<IdentityProvider> providers) {
+		this.providers = providers;
+	}
+
+	private List<Saml2RelyingPartyRegistration> getIdentityProviders(List<IdentityProvider> identityProviders) {
+		return identityProviders.stream()
+				.map(
+					p -> StringUtils.hasText(p.getLocalSpEntityIdTemplate()) ?
+							new Saml2RelyingPartyRegistration(
+								p.getEntityId(),
+								p.getAlias(),
+								p.getWebSsoUrlAsURI(),
+								p.getProviderCredentials(),
+								p.getLocalSpEntityIdTemplate()
+							) :
+							new Saml2RelyingPartyRegistration(
+								p.getEntityId(),
+								p.getAlias(),
+								p.getWebSsoUrlAsURI(),
+								p.getProviderCredentials()
+							)
+				)
+				.collect(Collectors.toList());
+	}
+
+	public static class IdentityProvider {
```

Comment:
Currently `RelyingPartyRegistration` does not need complex sub types. It could have been composed of a sub class called `ServiceProvider` with config locally, and another one called `IdentityProvider`. But for simplicity sake it was decided through previous reviews that `RelyingPartyRegistration` represents the combination of an SP/IDP pair.

A more complex structure would simplify future additions of properties, the question is would we over engineer it if those features were never needed or requested?